### DISCUSSION TITLE
Skip storing "Target"."Old" and "Environment" in the underlying store.

### DISCUSSION THREAD

**Discussion by buczkow** [thepirat000/Audit.NET#525] [Q&A] [answered]
Is there any way to narrow dataset stored in underlying store ?
Let's assume I would like to skip persisting bolded json:
{
    "EventType": "Order:Update",
    **"Environment": {
        "UserName": "Federico",
        "MachineName": "HP",
        "DomainName": "HP",
        "CallingMethodName": "Audit.UnitTest.AuditTests.TestUpdate()",
        "Exception": null,
        "Culture": "en-GB"
    },**
    "StartDate": "2016-08-23T11:33:14.653191-05:00",
    "EndDate": "2016-08-23T11:33:23.1820786-05:00",
    "Duration": 8529,
    "Target": {
        "Type": "Order",
        **"Old": {
            "OrderId": "39dc0d86-d5fc-4d2e-b918-fb1a97710c99",
            "Status": 2,
            "OrderItems": [{
                "Sku": "1002",
                "Quantity": 3.0
            }]
        },**
        "New": {
            "OrderId": "39dc0d86-d5fc-4d2e-b918-fb1a97710c99",
            "Status": -1,
            "OrderItems": null
        }
    }
}

**Comment by thepirat000** [comment, marked as answer]
One option is to set up a Custom Action to remove the data you don't need, for example:

```c#
Audit.Core.Configuration.AddOnSavingAction(scope =>
{
    scope.Event.Environment = null;
    if (scope.Event.Target?.Old != null)
    {
        scope.Event.Target.Old = null;
    }
});
```

If you are not using Target.Old, maybe it's better to just use a [Custom Field](https://github.com/thepirat000/Audit.NET#custom-fields-and-comments), for example to create a field named "Order":

```c#
using var scope = AuditScope.Create("Event Type", null, new { Order = order });
```

or 

```c#
using var scope = AuditScope.Create("Event Type", null);
scope.SetCustomField("Order", order);
```
